cmd/unused-exporter/exporter.go

```diff
@@ -286,10 +291,14 @@ func lastUsedTS(d unused.Disk) float64 {
 }
 
 func getRegionFromZone(p unused.Provider, z string) string {
-	if strings.ToLower(p.Name()) == "azure" {
-		return z
+	var region string
+	switch p.Name() {
+	case azure.ProviderName:
+		region = z
+	case gcp.ProviderName:
+		region = z[:strings.LastIndex(z, "-")]
+	case aws.ProviderName:
+		region = z[:len(z)-1]
 	}
-
-	// Drop the last character to get the region from the zone for GCP and AWS
-	return z[:len(z)-1]
+	return region
```

Comment:
Git doesn't allow me to comment in the whole change because, well, git, but I've a question: say that we rephrase this into:

```go
switch p.Name() {
case azure.ProviderName:
	return z
case gcp.ProviderName:
	return z[:strings.LastIndex(z, "-")]
case aws.ProviderName:
	return z[:len(z)-1]
default:
	panic("unrecognized provider name: " + p.Name())
}
```

My reasoning for doing this is that we are covering in the `switch` statement all three valid possibilities, and failure to conform to any of the `case` entries **should** be a bug, but kind of an unrecoverable one, that's why the `panic` and not returning an error.

Am I being too defensive here?

Reply:
I wondered the same. I'm ok with either. I feel like this would be more complete and a better practise, it kinda feels like what it should be... I also don't think we'll encounter that case 😅 but hey, so did the folks who used just two digits for the year, right?

Reply:
Hah that's a great example. Yeah, NOTHING seems to indicate that it would ever be possible, that's why I suggested a `panic` in the unlikely case.

Reply:
+1 panic()'ing wildo

Reply:
fixed at e23cc9b